src/steamcompmgr.cpp

```diff
@@ -2840,18 +2842,11 @@ steamcompmgr_main (int argc, char **argv)
 
 					if ( ev.xclient.data.l[0] == 24 && ev.xclient.data.l[1] == 8 )
 					{
-						// Message from vblankmanager
-						std::lock_guard<std::mutex> lock( lastvblankmsgtv_lock );
-
-						struct timeval vblankmsgreceivedtv = {};
-						gettimeofday( &vblankmsgreceivedtv, nullptr );
-
-						struct timeval timediff = {};
-
-						timersub( &vblankmsgreceivedtv, &lastvblankmsgtv, &timediff );
+						uint64_t vblankrecieved = get_time_in_nanos();
+						uint64_t diff = vblankrecieved - g_lastvblank;
```

Comment:
I'm not comfortable with making this subtraction use unsigned integers, I'd prefer to stick to signed integers. Or at least add an `assert(g_lastvblank <= vblankrecieved)`?

Reply:
MONOTONIC_RAW is well... monotonic. It cannot go back on itself.

It will only overflow every 584 years and I think at that we can call it a day.

Reply:
I before E except after C

Reply:
On further thought (and beyond just the typo) I think there's merit to checking g_lastvblank < vblankrecieved and not doing the compare if it is. The last vblank itself is atomic, but there's no expectating of atomicity between the vblank timestamp getting updated and the message getting sent/handled.

So you could have a coincidental race where we happen to handle a message super late in the steamcompmgr side because we were blocked for a long time, and the vblankmanager has _just_ updated the timestamp for the next vblank and not sent the message yet. In this case we should draw twice and pull two frames, as we could handle the next event as well. Really we should be encoding that uint64_t into the two data members of our user event and remove that horrible hack from existence, now that you're encoding time in a more convenient way than a timespec. Right?

Reply:
For this PR, we can just load `g_lastvblank` before getting `vblankrecieved` to fix the race.

Yes, we can encode the vblank time in the `repaintMsg`. I think that is better suited for a follow up PR though (for bisecting any potential issues mainly.)

I've gone ahead and fixed that as well as my typos.